### DISCUSSION TITLE
How to deploy express server written on Typescript

### DISCUSSION THREAD

**Discussion by weeebdev** [vercel/vercel#5328] [Help] [answered]
I have huge problems with defining `now.json` for deploying express server written on Typescript.
my file structure:
- dist (here all compiled js files go, e.g. app.js)
- src
  - app.ts (my express server)

how to define proper `now.json` for that? I tried functions, but I don't know which runtime to use. They didn't specify runtime name for node js.

My scripts:
-  "start": "node dist/app.js",
- "dev": "nodemon src/app.ts",
- "build": "tsc -p ."

**Comment by weeebdev** [comment]
Please, could you answer soon?) I am on a hackathon now, so I need to solve this problem))

**Comment by weeebdev** [comment, marked as answer]
Nvm, I moved to heroku

**Reply by kingRayhan** [reply]
is that a solution :laughing:

**Reply by amyegan** [reply]
We have a guide that might not have existed when this question was originally asked. You can learn more here: [Using Express.js with Vercel](https://vercel.com/guides/using-express-with-vercel)

**Reply by playerx** [reply]
I'm having the same problem to deploy Express+Typescript app on Vercel. It should be straightforward and common use case, but it's super complicated on Vercel. I've tried all types of combination with no success. the only way it worked was when I compile ts files to js and commit it on git, which is a wrong way. My goal is to push ts files, Vercel should get the source code, run build, and find my express app. It doesn't work and didn't manage to find a working example. The example you pointed is not with typescript, it's just js.

**Reply by playerx** [reply]
And btw I think about moving to Deno Deploy. They have a really good DX, it just works

**Reply by amyegan** [reply]
I'm sorry you're running into trouble with this @playerx. 😥 Configuration on these things can be tricky. 

I tried just now and was able to deploy a TypeScript Express server by combining the [Standalone Express](https://vercel.com/guides/using-express-with-vercel#standalone-express) example and a guide titled [How to deploy your Serverless Typescript function in a global CDN in seconds for free](https://vdelacou.medium.com/how-to-deploy-your-typescript-function-in-a-global-cdn-in-seconds-for-free-2b953397d91). My [example site](https://my-express-api.vercel.app/api) and [repo](https://github.com/amyegan/my-express-api) are public if you want to see them.

If you have a public repo or [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that you can share, I would be happy to take a look and help with debugging! 🐛
